src/main/java/stepsdesign/SysadminSteps.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package stepsdesign;
+
+import io.cdap.e2e.pages.actions.CdfSysAdminActions;
+import io.cdap.e2e.utils.CdfHelper;
+import io.cucumber.java.en.Then;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * CDF sysadmin related step design.
+ */
+public class SysadminSteps implements CdfHelper {
+
+  private static final Logger logger = LoggerFactory.getLogger(SysadminSteps.class);
+
+  @Then("Click on the {string} link from menu")
```

Comment:
Overall the changes looks good for this page we just may have to revisit the step definition naming approach.

For example I see definition namings like
- Select the Namespace from the System admin page (Can be updated to ```Select the Namespace```)
- Click on edit system preferences to set system preferences (Can be updated to ``` Click on edit system preferences```)
- Click on Delete button to delete the added preference and Click on Reset button to reset the added preferences
``` Delete the preferences and Reset the preferences```

These are just for idea, Step definition should clearly state what the step is doing. We may not have to provide additional information.

Discuss this internally once with the QA team, based on the discussion you can make the changes if required.

Reply:
The Navigations are different from sysadmin and hence used system admin for first step, for others changed as suggested.